cmd/otelcol/main.go

```diff
@@ -286,6 +287,25 @@ func setMemoryLimit(memTotalSizeMiB int) int {
 	return memLimit
 }
 
+// Sets environment variables expected by agent_config.yaml if missing
+func setDefaultEnvVars() {
+	realm := os.Getenv("SPLUNK_REALM")
+	token := os.Getenv("SPLUNK_ACCESS_TOKEN")
```

Comment:
realized that if these aren't set invalid urls would be used for the default env vars, which may still be used by a non-default config (the only case where realm and token would be allowed to be unset) and lead to ~hard to track down undesired behavior in edge cases.  Would probably be worth only adding the applicable inner arrays to `testArgs` if realm and then token are detected.